contracts/Exchanger.sol

```diff
@@ -2,49 +2,329 @@
 pragma solidity 0.8.13;
 
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IExchanger} from "./interfaces/IExchanger.sol";
+import {ICurveFiCurve} from "./interfaces/ICurveFi.sol";
+import {ParameterStore} from "./ParameterStore.sol";
 import "hardhat/console.sol";
 
-contract Exchanger {
-    /* Privileged functions: Admin */
+/// TODO cleanup unecessary variables like amountOUSD = amountOUSD
+/// TODO Make sure funds end up at coordinator address for WINDUP and UNWIND
+/// TODO Approval & Allownace should NOT BE MAX VALUES for pools
+/// Use the overloaded function with TO parameter for exchange
 
-    /// @dev initialize Vault
-    /// @param _tokenLvUSD lvUSD contract address
-    /// @param _tokenCoordinator Coordinator contract address
+/// @title Exchanger
+/// @dev is in charge of interacting with the CurveFi pools
+contract Exchanger is IExchanger {
+    using SafeERC20 for IERC20;
+
+    address internal _addressParameterStore;
+    address internal _addressCoordinator;
+    address internal _addressLvUSD;
+    address internal _addressOUSD;
+    address internal _address3CRV;
+    address internal _addressPoolLvUSD3CRV;
+    address internal _addressPoolOUSD3CRV;
+    IERC20 internal _lvusd;
+    IERC20 internal _ousd;
+    IERC20 internal _3crv;
+    ICurveFiCurve internal _poolLvUSD3CRV;
+    ICurveFiCurve internal _poolOUSD3CRV;
+
+    ParameterStore internal _paramStore;
+    int128 internal _indexLvUSD = 0;
+    int128 internal _indexOUSD = 0;
+    int128 internal _index3CRV = 1;
+
+    // TODO should slippage be in ParamStore?
```

Comment:
I dont think they should be in paramStore as they are not used any where else

Remove todo and we move it in the future if needed

Reply:
I created some Asana tickets to add these to ParamStore if we want later, so we dont forget. Going to remove the "TODO" in the code.